### PR TITLE
Fix broken documentation links

### PR DESCRIPTION
**Pull request by cbliard**
Also make some links more self-descriptive for accessibility reasons.

Remove trailing white spaces because my editor removes them automatically (and yours should too).

Broken links were found with markdown-link-check with the following command:
```
find . -name tmp -prune -o -name node_modules -prune -o -name \*.md -print0 | xargs -0 -n1 markdown-link-check
```

### CODE REVIEW

File: docs/installation-and-operations/installation/manual/README.md [hunk 1]
```diff
@@ -410,7 +410,7 @@ Restart the OpenProject server afterwards:
 
 The next web-request to the server will take longer (as the application is restarted). All subsequent request should be as fast as always.
 
-We encourage you to extend OpenProject yourself by writing a plug-in. Please, read the [plugin-contributions](https://community.openproject.org/projects/openproject/wiki/Developing_Plugins) guide for more information.
+We encourage you to extend OpenProject yourself by writing a plug-in. Please, read the [plugin creation guide](https://www.openproject.org/docs/development/create-openproject-plugin/) for more information.
```
Comment (as-op) [inline docs/installation-and-operations/installation/manual/README.md:413]: Just a tiny nickpick. It's nicer to have relative links, so they work on Github or local, too.

File: frontend/src/locales/README.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
 This folder will contain generated translations in your development or production OpenProject environment.
 
 To find out more how to contribute to translations,
-please see [our translation guide](^https://docs.openproject.org/development/#translations).
\ No newline at end of file
+please see [our translation guide](https://docs.openproject.org/development/#translations).
```
Comment (as-op) [inline frontend/src/locales/README.md:4]: The subdomain is no longer in use and redirects to https://www.openproject.org/docs/development/#translations

**Review by as-op — APPROVED** [review]
Thanks a lot for improving this!  ❤️

File: frontend/src/locales/README.md (hunk 2 above)
Comment (as-op) [inline frontend/src/locales/README.md]: The subdomain is no longer in use and redirects to https://www.openproject.org/docs/development/#translations


```suggestion
please see [our translation guide](https://www.openproject.org/docs/development/#translations).
```

File: docs/installation-and-operations/installation/manual/README.md (hunk 1 above)
Comment (as-op) [inline docs/installation-and-operations/installation/manual/README.md]: Just a tiny nickpick. It's nicer to have relative links, so they work on Github or local, too.

```suggestion
We encourage you to extend OpenProject yourself by writing a plug-in. Please, read the [plugin creation guide](../../../development/create-openproject-plugin/) for more information.
```

Comment (cbliard) [reply docs/installation-and-operations/installation/manual/README.md]: Thanks, that makes sense.
There are a lot of non-relative links remaining.

**Pull request merged**